src/database/DataTypes/RequisitionItem.js

```diff
@@ -230,6 +230,135 @@ export class RequisitionItem extends Realm.Object {
   get lastRequisitionDate() {
     return this.item?.lastRequisitionDate;
   }
+
+  resetClosingVariables() {
+    this.openingStock = 0;
+    this.outgoingStock = 0;
+    this.incomingStock = 0;
+    this.positiveAdjustments = 0;
+    this.negativeAdjustments = 0;
+  }
+
+  setIncomingStock(value) {
+    const newClosing =
+      this.openingStock +
+      value -
+      this.outgoingStock +
+      this.positiveAdjustments -
+      this.negativeAdjustments;
+
+    const canUpdate = newClosing >= 0;
+
+    if (canUpdate) {
+      this.incomingStock = value;
+    } else {
+      this.resetClosingVariables();
+    }
+
+    return canUpdate;
+  }
+
+  setOutgoingStock(value) {
+    const newClosing =
+      this.openingStock +
+      this.incomingStock -
+      value +
+      this.positiveAdjustments -
+      this.negativeAdjustments;
+
+    const canUpdate = newClosing >= 0;
+
+    if (canUpdate) {
+      this.outgoingStock = value;
+    } else {
+      this.resetClosingVariables();
+    }
+
+    return canUpdate;
+  }
+
+  setPositiveAdjustments(value) {
+    const newClosing =
+      this.openingStock +
+      this.incomingStock -
+      this.outgoingStock +
+      value -
+      this.negativeAdjustments;
+
+    const canUpdate = newClosing >= 0;
+
+    if (canUpdate) {
+      this.positiveAdjustments = value;
+    } else {
+      this.resetClosingVariables();
+    }
+
+    return canUpdate;
+  }
+
+  setNegativeAdjustments(value) {
+    const newClosing =
+      this.openingStock +
+      this.incomingStock -
+      this.outgoingStock +
+      this.positiveAdjustments -
+      value;
+
+    const canUpdate = newClosing >= 0;
+
+    if (canUpdate) {
+      this.negativeAdjustments = value;
+    } else {
+      this.resetClosingVariables();
+    }
+
+    return canUpdate;
+  }
+
+  get closingStock() {
```

Comment:
Having a look at `mSupply` and this should actually be saved and equal to `stockOnHand`. I think the two columns is fine/good but it's should be a duplicated column rather than recalculating